src/librustc/ty/context.rs

```diff
@@ -288,6 +288,18 @@ pub struct ResolvedOpaqueTy<'tcx> {
     pub substs: SubstsRef<'tcx>,
 }
 
+/// Stores the type, span and optional scope span of all types that are live
```

Comment:
Whenever a value may be live across a generator yield, the type of that value winds up in the `GeneratorInterior` struct. This struct adds additional information about such captured types that can be useful for diagnostics. In particular, it stores the span that caused a given type to be recorded, along with the scope that enclosed the value (which can be used to find the await that the value is live across).

For example:

```
async move {
    let x: T = ...;
    foo.await
    ...
}
```

Here, we would store the type `T`, the span of the value `x`, and the "scope-span" for the scope that contains `x`.

Reply:
Fixed!